### PR TITLE
Run with google-cloud-sdk

### PR DESCRIPTION
**Pull request by bgpat**
## WHY

- server dies suddenly
- searching packages cannot work

## WHAT

Use google-cloud-sdk following [official wiki](https://github.com/golang/gddo/wiki/Development-Environment-Setup).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+	yes | /google-cloud-sdk/install.sh && \
+	/google-cloud-sdk/bin/gcloud components install app-engine-python && \
+	sed -ie 's/"redis:\/\/127\.0\.0\.1:6379"/os.Getenv("REDIS_URL")/' /go/src/github.com/golang/gddo/database/database.go && \
+	apk del git && rm -f /google-cloud-sdk.tar.gz
+
+ADD run.sh /run.sh
 
 EXPOSE 8080
 
-CMD /go/bin/gddo-server -db-server=$REDIS_URL
```
Comment (koudaiii) [inline Dockerfile]: Is `/google-cloud-sdk/bin/dev_appserver.py --host 0.0.0.0 $APP_DIR` include `/go/bin/gddo-server` ?

Comment (bgpat) [inline Dockerfile]: Yes, it's [here (run.sh)](https://github.com/wantedly/godoc/pull/5/files/8581402fc29cfc919fddaaa76eec71fab7208fc0#diff-1b0c2b516b83393edb7200ad5ff12181R17).

Comment (bgpat) [inline Dockerfile]: No, `/go/bin/gddo-server` is not used.

Comment (koudaiii) [inline Dockerfile]: ok

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,16 @@
-# Official document say: Install Go1.6.
-# https://github.com/golang/gddo/wiki/Development-Environment-Setup
-FROM golang:1.6-alpine
+FROM golang:1.8-alpine
```
Comment (koudaiii) [inline Dockerfile]: if golang is not use, I think change python.
Is my understanding correct?

Comment (bgpat) [inline Dockerfile]: `dev_appserver.py` uses golang internal.

Comment (koudaiii) [inline Dockerfile]: ❗️ 
ok

File: run.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+APP_DIR=/go/src/github.com/golang/gddo/gddo-server
+APP_YAML=$APP_DIR/app.yaml
+
+if [ -e $APP_YAML.bak ]; then
```
Comment (koudaiii) [inline run.sh]: What is the pattern that `$APP_YAML.bak` exists? 👀

File: run.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	cp -f $APP_YAML.bak $APP_YAML
+else
+	cp $APP_YAML $APP_YAML.bak
+fi
+
+echo 'env_variables:' >> $APP_YAML
+env | sed -E 's/^(.*?)=(.*)$/  \1: "\2"/' >> $APP_YAML
```
Comment (koudaiii) [inline run.sh]: Can you tell me the output example of this? 👁

File: run.sh (hunk 3 above)
Comment (bgpat) [inline run.sh]: To add [env_variables](https://cloud.google.com/appengine/docs/standard/python/config/appref#Python_app_yaml_Defining_environment_variables).

File: run.sh (hunk 4 above)
Comment (bgpat) [inline run.sh]:
```
echo 'env_variables:'; env | sed -E 's/^(.*?)=(.*)$/  \1: "\2"/'
env_variables:
  GITHUB_TOKEN: "bbaacca5c1bf9db4dc364c31370cf21d30848933"
  no_proxy: "*.local, 169.254/16"
  HOSTNAME: "dc54bc632b8a"
  SHLVL: "1"
  HOME: "/root"
  GOLANG_SRC_URL: "https://golang.org/dl/go1.8.src.tar.gz"
  BASIC_AUTH_USERNAME: ""
  TERM: "xterm"
  PATH: "/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
  GOPATH: "/go"
  PWD: "/go"
  GOLANG_SRC_SHA256: "406865f587b44be7092f206d73fc1de252600b79b3cacc587b74b5ef5c623596"
  REDIS_URL: "redis://redis:6379"
  GOLANG_VERSION: "1.8"
  BASIC_AUTH_PASSWORD: ""
```

Comment (bgpat) [inline run.sh]: echo 'env_variables:'; env | sed -E 's/^(.*?)=(.*)$/  \1: "\2"/'
env_variables:
  GITHUB_TOKEN: "****"
  no_proxy: "*.local, 169.254/16"
  HOSTNAME: "dc54bc632b8a"
  SHLVL: "1"
  HOME: "/root"
  GOLANG_SRC_URL: "https://golang.org/dl/go1.8.src.tar.gz"
  BASIC_AUTH_USERNAME: ""
  TERM: "xterm"
  PATH: "/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
  GOPATH: "/go"
  PWD: "/go"
  GOLANG_SRC_SHA256: "406865f587b44be7092f206d73fc1de252600b79b3cacc587b74b5ef5c623596"
  REDIS_URL: "redis://redis:6379"
  GOLANG_VERSION: "1.8"
  BASIC_AUTH_PASSWORD: ""

File: run.sh (hunk 3 above)
Comment (koudaiii) [inline run.sh]: This application run on container. Therefore  when `docker run` , `$APP_YAML.bak` is cleared.

I think I do not need this `if [ -e $APP_YAML.bak ]; then` 👀 
Do you think ?

File: run.sh (hunk 4 above)
Comment (koudaiii) [inline run.sh]: thanks 👍

File: run.sh (hunk 3 above)
Comment (bgpat) [inline run.sh]: It add `env_variables` to `app.yaml` twice on restarting server.
Restarted server don't work without this.

Comment (koudaiii) [inline run.sh]: wao. ok 👍

**Comment by koudaiii** [comment]
LGTM!!

**Pull request merged**